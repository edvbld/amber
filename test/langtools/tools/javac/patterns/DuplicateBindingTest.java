/*
 * @test /nodynamiccopyright/
 * @summary Basic pattern bindings scope test
 * @compile/fail/ref=DuplicateBindingTest.out -XDrawDiagnostics DuplicateBindingTest.java
 */

public class DuplicateBindingTest {

    int f;

    public static void main(String[] args) {

        if (args != null) {
            int s;
            if (args[0] instanceof String s) { // NOT OK. Redef same scope.
            }
            if (args[0] instanceof String f) { // OK to redef field.
            }
        }
    }
}
