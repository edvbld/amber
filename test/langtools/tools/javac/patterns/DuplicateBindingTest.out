DuplicateBindingTest.java:15:36: compiler.err.already.defined: kindname.variable, s, kindname.method, main(java.lang.String[])
DuplicateBindingTest.java:19:22: compiler.err.already.defined: kindname.variable, k, kindname.method, main(java.lang.String[])
DuplicateBindingTest.java:21:22: compiler.err.already.defined: kindname.variable, args, kindname.method, main(java.lang.String[])
DuplicateBindingTest.java:18:19: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: java.lang.String, java.lang.Integer)
4 errors
